### PR TITLE
[coq] Add "-q" to default flags.

### LINKED ISSUE #3924: [Coq] Use the flag -q by default
If I write a standard dune file to compile some coq developments, Coq is invoked without the `-q` argument, causing it to load a `coqrc` file. This reduces the reproducibility of the build since I don't know what users might have stored there. Note that `coq_makefile` also includes `-q` by default. Currently, I can use `(flags -q)` to get this behavior, but I think it would be good for `-q` to be an overridable default. Basically every Coq package would benefit from setting this option.

### PR DESCRIPTION
**Pull request by ejgallego**
Fixes #3924 , and a bug where `:standard` was not properly set in
theory flags, as they should be inherited from the ones set in the
profile; thanks Rudi Gringerb.

### CODE REVIEW

**Comment by rgrinberg** [comment]
> Actually this PR highlights a problem in the way we handle flags from
profile and from the theory stanza, they should form a single set I
think, however they are disjoint.

I'd try to match the OCaml behavior. Which would mean that `:standard` would be the `env` flags when expanding the theory flags.

File: CHANGES.md
Comment (rgrinberg) [inline CHANGES.md:72]: Should be #3931

**Comment by ejgallego** [comment]
> I'd try to match the OCaml behavior. Which would mean that `:standard` would be the `env` flags when expanding the theory flags.

Of course, thanks; I was a bit dense writing this PR.

**Comment by ejgallego** [comment]
@rgrinberg , this should be ready. I'll wait for you to have quick look before merge.

**Comment by rgrinberg** [comment]
Looks good. I'd recommend to:

* Add a test
* Describe this change in the CHANGES list as well.

**Comment by ejgallego** [comment]
> * Add a test

Indeed testing would be good but I am not sure how to reliable get the flags, `_build/log` does contain path-dependent data.

> Describe this change in the CHANGES list as well

Isn't that done? Is there some other file I should update?

**Comment by rgrinberg** [comment]
The CHANGES only mention that you've added `-q` to the default set of flags. You didn't mention that you've changed `:standard` to mean all currently set flags. Previously, `:standard` was always empty.

**Comment by ejgallego** [comment]
> The CHANGES only mention that you've added `-q` to the default set of flags. You didn't mention that you've changed `:standard` to mean all currently set flags. Previously, `:standard` was always empty.

Oh indeed, done thanks. Updated doc in the way.

**Comment by rgrinberg** [comment]
@ejgallego anything blocking this PR?

**Comment by ejgallego** [comment]
@rgrinberg I'm unsure how to add a test for this, otherwise it seems ready.

> Indeed testing would be good but I am not sure how to reliable get the flags, `_build/log` does contain path-dependent data

**Comment by rgrinberg** [comment]
it's fine to merge this without the tests if you've verified that it works as expected

**Review by rgrinberg — APPROVED** [review]

**Comment by ejgallego** [comment]
Thanks @rgrinberg , will have a last look at a possible test and merge.

**Comment by ejgallego** [comment]
> will have a last look at a possible test and merge.

Didn't find a nice way for this particular options. At some point we should strengthen the flags test suite for Coq itself, but IMHO this is good to go.

**Pull request merged**
